hedgehog/src/Hedgehog/Internal/Shrink.hs

```diff
@@ -31,10 +31,9 @@ towards destination x =
     let
       -- Halve the operands before subtracting them so they don't overflow.
       -- Consider 'minBound' and 'maxBound' for a fixed sized type like 'Int64'.
-      diff =
-        (x `quot` 2) - (destination `quot` 2)
+      diff = (toInteger x `quot` 2) - (toInteger destination `quot` 2)
```

Comment:
>This fixes the odd situation in which `2` is not representable as a number in a type. In particular this occurs in the package *clash-prelude*, which defines types such as `Unsigned 1` where only `0` and `1` can be represented.

This looks reasonable, though I was wondering if we can tell the bounds of the type instead of `toInteger` ?

Reply:
I don't think so - not without adding extra type class constraints (and there is the inevitable problem that `Integer` itself has no bounds)

Reply:
I think you detect it by roundtripping `2` through `fromInteger/toInteger`, the question is if GHC is smart enough to make that a constant.